pkg/envoy/proxy.go

```diff
@@ -198,5 +221,7 @@ func NewProxy(certCommonName certificate.CommonName, certSerialNumber certificat
 		lastNonce:          make(map[TypeURI]string),
 		lastSentVersion:    make(map[TypeURI]uint64),
 		lastAppliedVersion: make(map[TypeURI]uint64),
+
+		lastxDSResourcesSent: make(map[TypeURI]mapset.Set),
```

Comment:
I wonder if this should be a sync.Map? 
Do we need to worry about concurrency?

Reply:
I see that you are using `mapset` to do a set operation - so this makes sense!
I'd probably leave a comment here to show what the Set way look like -- a) it is a string and b) what kind of strings